pkg/trtl/trtl_test.go

```diff
@@ -1,39 +1,323 @@
+// should clean up.
+func generateDB(s *trtlTestSuite) {
+	require := s.Require()
+	db, err := leveldb.OpenFile(s.db, nil)
+	require.NoError(err)
+	defer db.Close()
+
+	loadFixtures(s)
+	fmt.Println(dbFixtures["alice"])
+
+	// Write all the test fixtures to the database.
+	for _, fixture := range dbFixtures {
+		// Must be wrapped in a honu object to be retrievable with honu.
+		meta := &object.Object{
+			Key:       []byte(fixture.Key),
+			Namespace: fixture.Namespace,
+			Region:    metaRegion,
+			Owner:     metaOwner,
+			Version:   metaVersion,
+		}
+		meta.Data, err = json.Marshal(fixture.Value)
+		require.NoError(err)
+		data, err := proto.Marshal(meta)
+		require.NoError(err)
+		err = db.Put([]byte(fixture.Namespace+"::"+fixture.Key), data, nil)
+		require.NoError(err)
+	}
+
+	writeGzip(s)
+	log.Info().Msg("successfully regenerated test fixtures")
+}
+
+// extractGzip extracts the gzipped database to the temporary directory.
+func extractGzip(s *trtlTestSuite) {
+	// Read the gzip file.
+	require := s.Require()
+	f, err := os.Open(s.gzip)
+	require.NoError(err)
+	defer f.Close()
+	gr, err := gzip.NewReader(f)
+	require.NoError(err)
+	defer gr.Close()
+
+	// Write the contents to a directory.
+	tr := tar.NewReader(gr)
+	for {
+		hdr, err := tr.Next()
+		if err == io.EOF {
+			break
+		}
+		require.NoError(err)
+		switch hdr.Typeflag {
+		case tar.TypeDir:
+			err = os.MkdirAll(filepath.Join(s.db, hdr.Name), os.FileMode(hdr.Mode))
+			require.NoError(err)
+		case tar.TypeReg:
+			f, err := os.Create(filepath.Join(s.db, hdr.Name))
+			require.NoError(err)
+			_, err = io.Copy(f, tr)
+			require.NoError(err)
+			f.Close()
+		default:
+			require.Fail(fmt.Sprintf("extracting %s: unexpected type: %v", hdr.Name, hdr.Typeflag))
+		}
+	}
+}
+
+// writeGzip writes the database in the temporary directory to a gzipped file.
+func writeGzip(s *trtlTestSuite) {
+	require := s.Require()
+	// Create a gzip file.
+	f, err := os.Create(s.gzip)
+	require.NoError(err)
+	defer f.Close()
+	w := gzip.NewWriter(f)
+	defer w.Close()
+
+	// Create a tar file.
+	tw := tar.NewWriter(w)
+	defer tw.Close()
+
+	// Write the DB to the tar file.
+	err = filepath.Walk(s.db, func(path string, info os.FileInfo, err error) error {
+		require.NoError(err)
+		hdr, err := tar.FileInfoHeader(info, "")
+		require.NoError(err)
+		hdr.Name = path[len(s.db):]
+		err = tw.WriteHeader(hdr)
+		require.NoError(err)
+		if info.IsDir() {
+			return nil
+		}
+		f, err := os.Open(path)
+		require.NoError(err)
+		defer f.Close()
+		_, err = io.Copy(tw, f)
+		require.NoError(err)
+		return nil
+	})
+	require.NoError(err)
+}
+
+func (s *trtlTestSuite) SetupSuite() {
+	var err error
+	require := s.Require()
+	s.gzip = filepath.Join("testdata", "db.tar.gz")
+	s.db, err = ioutil.TempDir("testdata", "db*")
+	require.NoError(err)
+
+	// Only regenerate the test database if requested or it doesn't exist.
+	if _, err = os.Stat(s.gzip); *update || os.IsNotExist(err) {
+		generateDB(s)
```

Comment:
since there's a lot going on in the `generateDB` function, would you mind adding a comment here to remind us that it also calls `loadFixtures` under the hood?